security-force-monitor/sfm-cms
# Update models and importer for field harmonization

## Overview

Update the models and the importer to reflect new and removed fields as part of the field harmonization effort.

Connects #657, #658.

## Testing Instructions

This PR requires you to create a new development environment with Docker and run a fresh import. This process can take up to an hour, so plan to have something else to work on while the import is running.

* Copy the new dev settings over:

```
blackbox_cat configs/settings_local_dev.py.gpg > sfm_pc/settings_local.dev.py
```

* Build Docker containers with `docker-compose build`
* Start the database service with `docker-compose up postgres`
* Get `sfm_osm_data.dump` from me, copy it to your `sfm-cms` directory, and in another shell run:

```
docker-compose run --rm app pg_restore -h postgres -U postgres -d sfm -O sfm_osm_data.dump
```

* Run migrations with `docker-compose run --rm migration`
* Get `location_models.json` from me, copy it to your `sfm-cms` directory, and run:

```
docker-compose run --rm app ./manage.py loaddata location_models.json
```

* If you have existing `auth_models.json`, load them too:

```
docker-compose run --rm app ./manage.py loaddata auth_models.json
```

* Quit the `postgres` service that you brought up earlier and bring up the full development environment with `docker-compose up`
* In another shell, run the import with the new testing data dump (it's OK if this prints some warnings, but the process should complete without raising a runtime error):

```
docker-compose run --rm app ./manage.py import_google_doc --doc_id 1GoC8TqV3b62-CD6oEcRwMzVwygFs0P0H6YswvRZt71M --source_doc_id 1z8m-7YmK4BdqAWC9t4UVfrMaJJxR_64dBndRSb5MYps
```

* Update your `Country` models with `docker-compose run --rm app ./manage.py update_countries_plus`
* Recreate your search index with `docker-compose run --rm app ./manage.py make_search_index --recreate`
* Once the search index has finished updating, check that the following public-facing pages work properly:
    * Search views for all entity types
    * Detail views for all entity types
    * Data downloads page
    * (Note that the `Countries` page won't load properly since this is only a subset of the data that does not include certain entities linked to on that page)
* Clean up your `Location` file with `rm location_models.json`
